File: Haskell/Atividade-5/exercicio-12.hs
{-
Crie uma funcao com assinatura apagar :: Int -> [t] -> [t], a qual recebe um numero de elementos, uma lista, e retorna uma lista.
Esta funcao deve remover da lista os n primeiros elementos fornecidos como paramentro.
-}

apagar :: Int -> [t] -> [t]
apagar _ [] = []
apagar 1 (a:b) = b
apagar x (a:b) = (apagar (x - 1) b)

main = do
  print (apagar 3 [1..20])

File: Prolog/Atividade-16/exercicio-12.pl
/*
Crie uma regra apagar(N,L1,L2), a qual recebe um numero de elementos N, uma lista N1, e retorna uma lista L2. Esta funcao deve remover da lista os N primeiros elementos fornecidos como parametro.
*/

apagar(0,L,L).
apagar(N,[_|T],L2) :- N1 is N - 1, apagar(N1,T,L2).
